### PR TITLE
Wipe before reset/recovery and introduce 'initialized' field

### PR DESCRIPTION
**Pull request by tsusanka**
Replaces #1071.

Keeping the version field unchanged simplifies things. Most of this PR is moving the `is_initialized` function to device.py which is not needed if you oppose. It seemed clearer to me because the _device_ is initialized and not just the storage.

Anyway, the most important part now is in the `init_unlocked`:

```
if device.is_version_stored() and not device.is_initialized():
    common.set_bool(common.APP_DEVICE, device.INITIALIZED, True, public=True)
```

I think this works just fine, but I am still wondering if we should not introduce a new storage version just to be on the safe side (and include this snippet in the migration).

### CODE REVIEW

**Comment by matejcik** [comment]
> ```
> if device.is_version_stored() and not device.is_initialized():
>     common.set_bool(common.APP_DEVICE, device.INITIALIZED, True, public=True)
> ```
> 
> I think this works just fine, but I am still wondering if we should not introduce a new storage version just to be on the safe side (and include this snippet in the migration).

i don't think we need to do that. we are not changing meaning of any fields.

File: core/src/storage/__init__.py [hunk 1]
```diff
@@ -20,6 +16,14 @@ def init_unlocked() -> None:
     version = device.get_version()
     if version == common.STORAGE_VERSION_01:
         _migrate_from_version_01()
+    if device.is_version_stored() and not device.is_initialized():
```
Comment (andrewkozlik) [inline core/src/storage/__init__.py:19]: A comment here would be helpful and whitespace to separate this block from "# Check for storage version upgrade".

File: core/src/storage/device.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
 _SLIP39_IDENTIFIER         = const(0x10)  # bool
 _SLIP39_ITERATION_EXPONENT = const(0x11)  # int
 _SD_SALT_AUTH_KEY          = const(0x12)  # bytes
+INITIALIZED               = const(0x13)  # bool (0x01 or empty)
```
Comment (andrewkozlik) [inline core/src/storage/device.py:36]: nit: whitespace

File: core/src/storage/__init__.py (hunk 1 above)
Comment (tsusanka) [reply core/src/storage/__init__.py:22]: Thanks, fixed in afd8acb25f31dc8d8cedc15576f3b37262f7a455.

File: core/src/storage/device.py (hunk 2 above)
Comment (tsusanka) [reply core/src/storage/device.py]: Thanks, fixed in afd8acb25f31dc8d8cedc15576f3b37262f7a455.

File: core/src/storage/__init__.py [hunk 3]
```diff
@@ -20,6 +16,14 @@ def init_unlocked() -> None:
...
+        common.set_bool(common.APP_DEVICE, device.INITIALIZED, True, public=True)
+
+
+def reset() -> None:
```
Comment (matejcik) [inline core/src/storage/__init__.py:26]: please add a docstring with a brief explanation of what's going on here

Comment (tsusanka) [reply core/src/storage/__init__.py:26]: Fixed.

**Pull request merged**